app/build_images.go

```diff
@@ -0,0 +1,225 @@
+package app
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"code.cloudfoundry.org/fissile/builder"
+	"code.cloudfoundry.org/fissile/docker"
+	"code.cloudfoundry.org/fissile/model"
+	"github.com/SUSE/stampy"
+	"github.com/fatih/color"
+)
+
+// BuildImagesOptions contains all option values for the `fissile build images` command.
+type BuildImagesOptions struct {
+	Force                    bool
+	Labels                   map[string]string
+	NoBuild                  bool
+	OutputDirectory          string
+	PatchPropertiesDirective string
+	Roles                    []string
+	Stemcell                 string
+	StemcellID               string
+	TagExtra                 string
+}
+
+// BuildImages builds all role images using releases
+func (f *Fissile) BuildImages(opt BuildImagesOptions) error {
+	err := f.LoadManifest()
+	if err != nil {
+		return err
```

Comment:
I don't know; I find 500 character error messages (because each call level added some prefix to it) rather tedious. In that case I would rather have a full stacktrace.

In this particular example, the only possible error return is already prefixed with `Error loading role manifest: %s`: https://github.com/cloudfoundry-incubator/fissile/blob/a19161d/app/fissile.go#L71-L73

I don't see how putting another prefix that is totally unrelated to the failure, but just *hints* at the call chain improves this. It is kind of obvious that the image wasn't built when the build aborted because it didn't load the manifest.